dftimewolf/metawolf/metawolf.py

```diff
@@ -0,0 +1,802 @@
+            # Session and recipe settables are not in scope.
+            continue
+          value = input('Value for parameter {0:s} ({1:s}: {2:s}. Press '
+                        '"enter" to skip.): '.format(
+              settable.name,
+              self.metawolf_output.Color('Current value', output.YELLOW),
+              self.metawolf_output.Color(settable.GetValue(), output.GREEN)))
+          value = str(value).strip()
+          if not value:
+            continue
+          updated = self.UpdateSessionSettable(value, settable=settable)
+          while not updated:
+            value = input('Value for parameter {0:s}: '.format(settable.name))
+            value = str(value).strip()
+            updated = self.UpdateSessionSettable(value, settable=settable)
+      return
+
+    if what[0] == SESSION_ID_SETTABLE:
+      session_id = what[1]
+      if session_id not in self.sessions:
+        self.poutput(self.metawolf_output.Color(
+            'Session {0:s} does not exist.'.format(session_id), output.RED))
+        return
+
+      self.SetSessionID(session_id)
+      last_active = self.sessions.get(session_id, {}).get(LAST_ACTIVE_RECIPE)
+      if last_active:
+        self.recipe_settable.SetValue(last_active)
+      else:
+        self.recipe_settable.SetValue(None)
+        self.ClearSessionSettables()
+      self.reload_settables = True
+      return
+
+    if what[0] == RECIPE_SETTABLE:
+      recipe = what[1]
+      if recipe not in self.recipes:
+        self.poutput(self.metawolf_output.Color(
+            'Recipe {0:s} does not exist.'.format(recipe), output.RED))
+        return
+
+      if self.recipe:
+        # If a recipe is already set, prompt the user for copy over option
+        value = input('Would you like to copy over previous recipe ({0:s}) '
+                      'arguments to the new recipe ({1:s}): [yN]? '.format(
+            self.recipe, recipe)) or 'n'
+        copy_over = self.metawolf_utils.Str2Bool(str(value))
+        while copy_over not in [False, True]:
+          value = input('[yN]? ') or 'n'
+          copy_over = self.metawolf_utils.Str2Bool(str(value))
+        self.copy_over = copy_over
+
+      if self.recipe != recipe:
+        # Recipe changed, we need to reload the settables.
+        self.reload_settables = True
+
+      self.recipe_settable.SetValue(recipe)
+    else:
+      # This block happens whenever a recipe's argument / unknown argument
+      # is set
+      s_id = None
+      if self.session_id:
+        if self.recipe:
+          s_id = '{0:s}-{1:s}-{2:s}'.format(
+              self.session_id, self.recipe, what[0])
+        else:
+          if what[0] != SESSION_ID_SETTABLE:
+            self.poutput('You must set a recipe first: `{0:s}`'.format(
+                self.metawolf_output.Color('set recipe name', output.YELLOW)))
+          return
+      if s_id in self.session_settables:
+        _ = self.UpdateSessionSettable(what[1], s_id=s_id)
```

Comment:
It's `recipe` if `what[0] == RECIPE_SETTABLE`. At this point in code, `what[0]` holds the value of a settable parameter (e.g. `incident_id`). Since it's in the `else` clause, `recipe` is not defined (and also the variable name would be misleading there, as what it really is is some parameter's value)